src/components/editor/FileSelector.tsx

```diff
@@ -1,51 +1,97 @@
-import { List, ListItemButton, ListItemText } from '@mui/material';
+import { ListItemText } from '@mui/material';
+import {
+    DataGrid,
+    GridColDef,
+    GridRenderCellParams,
+    GridSelectionModel,
+} from '@mui/x-data-grid';
 import { LiveSpecQuery } from 'components/capture/details';
 import { EditorStoreState, useZustandStore } from 'components/editor/Store';
 import { DraftSpecQuery } from 'hooks/useDraftSpecs';
+import { useEffect, useState } from 'react';
 import { FormattedMessage } from 'react-intl';
 
-function EditorFileSelector() {
-    const currentCatalog = useZustandStore<
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>,
-        EditorStoreState<LiveSpecQuery | DraftSpecQuery>['currentCatalog']
-    >((state) => state.currentCatalog);
+const initialState = {
+    columns: {
+        columnVisibilityModel: {
+            spec_type: false,
+        },
+    },
+};
+
+const getRowId = (spec: any) => {
+    let newSelectionModel;
+
+    if (spec.id) {
+        newSelectionModel = spec.id;
+    } else if (spec.draft_id) {
+        newSelectionModel = spec.draft_id;
+    }
+
+    return `${newSelectionModel}-${spec.catalog_name}`;
+};
+
+const columns: GridColDef[] = [
+    {
+        field: 'catalog_name',
+        headerName: 'Files',
+        flex: 1,
+        renderCell: (params: GridRenderCellParams<Date>) => (
+            <ListItemText
+                primary={params.row.catalog_name}
+                secondary={params.row.spec_type}
+            />
+        ),
+    },
+    {
+        field: 'spec_type',
+        headerName: 'Type',
+    },
```

Comment:
We only display one column. Including type as a column so that the user can filter on it.